lib/WP_Auth0_SocialAmplification_Widget.php

```diff
@@ -16,30 +17,66 @@ function __construct() {
 			__( $this->getWidgetName(), 'wp_auth0_widget_domain' ),
 			array( 'description' => __( $this->getWidgetDescription(), 'wpb_widget_domain' ) )
 		);
+		$this->options = WP_Auth0_Options::Instance();
 	}
 
 	protected function getWidgetId() {
 		return 'wp_auth0_social_amplification_widget';
 	}
 
 	protected function getWidgetName() {
-		return "Auth0 Social Amplification";
+		return __( 'Auth0 Social Amplification', 'wp-auth0' );
+
 	}
 
 	protected function getWidgetDescription() {
-		return "Shows Auth0 Social Amplification Widget Embed in your sidebar";
+		return __( 'Shows Auth0 Social Amplification widget in a sidebar', 'wp-auth0' );
 	}
 
+	/**
+	 * Build the widget settings form in the Customizer and wp-admin > Widgets page.
+	 *
+	 * @param array $instance - Current instance of this widget.
+	 */
 	public function form( $instance ) {
-		require WPA0_PLUGIN_DIR . 'templates/a0-widget-amplificator.php';
+		$fields = array(
+			'amplificator_title' => __( 'Widget title', 'wp-auth0' ),
+			'amplificator_subtitle' => __( 'Widget subtitle', 'wp-auth0' ),
+			'social_twitter_message' => __( 'Twitter message', 'wp-auth0' ),
+		);
+
+		foreach ( $fields as $field => $title ) {
+			$field_value = isset( $instance[ $field ] )
+				? $instance[ $field ]
+				: $this->options->get( $field );
+			printf(
+				'<p><label for="%s">%s:</label><textarea class="widefat" id="%s" name="%s">%s</textarea></p>',
+				esc_attr( $this->get_field_id( $field ) ),
+				sanitize_text_field( $title ),
+				esc_attr( $this->get_field_id( $field ) ),
+				esc_attr( $this->get_field_name( $field ) ),
+				esc_textarea( $field_value )
+			);
+		}
+
+		printf(
+			'<p>%s <a href="%s" target="_blank">%s</a>.</p>
+			<p>%s</p><ul><li><code>%s</code> - %s</li><li><code>%s</code> - %s</li></ul>',
+			__( 'App keys are required and are set on the Advanced tab of the', 'wp-auth0' ),
+			admin_url( 'admin.php?page=wpa0#advanced' ),
+			__( 'Auth0 settings page', 'wp-auth0' ),
+			__( 'You can use the following tags in the messages above:', 'wp-auth0' ),
+			'%page_url%',
+			__( 'This will be replaced by the current page URL', 'wp-auth0' ),
+			'%site_url%',
+			__( 'This will be replaced by the site URL', 'wp-auth0' )
+		);
 	}
 
 	public function update( $new_instance, $old_instance ) {
-		$options = WP_Auth0_Options::Instance();
-		$options->set( 'social_facebook_message', $new_instance['social_facebook_message'] );
-		$options->set( 'social_twitter_message', $new_instance['social_twitter_message'] );
-		$options->set( 'amplificator_title', $new_instance['amplificator_title'] );
-		$options->set( 'amplificator_subtitle', $new_instance['amplificator_subtitle'] );
+		$new_instance['social_twitter_message'] = sanitize_text_field( $new_instance['social_twitter_message'] );
```

Comment:
Need to set this for the widget instance, not in a global setting (WordPress Customizer stops too many options from being set ... probably a security measure?)